pkg/config/remote/uptane/client.go

```diff
@@ -0,0 +1,245 @@
+	if err != nil {
+		return nil, err
+	}
+	buffer := &bufferDestination{}
+	err = c.configTUFClient.Download(path, buffer)
+	if err != nil {
+		return nil, err
+	}
+	return buffer.Bytes(), nil
+}
+
+func (c *Client) updateRepos(response *pbgo.LatestConfigsResponse) error {
+	err := c.targetStore.storeTargetFiles(response.TargetFiles)
+	if err != nil {
+		return err
+	}
+	c.directorRemoteStore.update(response)
+	c.configRemoteStore.update(response)
+	_, err = c.directorTUFClient.Update()
+	if err != nil {
+		return errors.Wrap(err, "could not update director repository")
+	}
+	_, err = c.configTUFClient.Update()
+	if err != nil {
+		return errors.Wrap(err, "could not update config repository")
+	}
+	return nil
+}
+
+func (c *Client) pruneTargetFiles() error {
+	targetFiles, err := c.directorTUFClient.Targets()
+	if err != nil {
+		return err
+	}
+	var keptTargetFiles []string
+	for target := range targetFiles {
+		keptTargetFiles = append(keptTargetFiles, target)
+	}
+	return c.targetStore.pruneTargetFiles(keptTargetFiles)
+}
+
+func (c *Client) verify() error {
+	err := c.verifyOrgID()
+	if err != nil {
+		return err
+	}
+	return c.verifyUptane()
+}
+
+func (c *Client) verifyOrgID() error {
+	directorTargets, err := c.directorTUFClient.Targets()
+	if err != nil {
+		return err
+	}
+	for targetPath := range directorTargets {
+		if !strings.HasPrefix(targetPath, c.orgIDTargetPrefix) {
+			return fmt.Errorf("director target '%s' does not have the correct orgID", targetPath)
+		}
+	}
+	return nil
+}
+
+func (c *Client) verifyUptane() error {
+	directorTargets, err := c.directorTUFClient.Targets()
+	if err != nil {
+		return err
+	}
+	for targetPath, targetMeta := range directorTargets {
+		configTargetMeta, err := c.configTUFClient.Target(targetPath)
+		if err != nil {
+			return fmt.Errorf("failed to find target '%s' in config repository", targetPath)
+		}
+		if configTargetMeta.Length != targetMeta.Length {
+			return fmt.Errorf("target '%s' has size %d in directory repository and %d in config repository", targetPath, configTargetMeta.Length, targetMeta.Length)
+		}
+		if len(targetMeta.Hashes) == 0 {
+			return fmt.Errorf("target '%s' no hashes in the director repository", targetPath)
+		}
+		if len(targetMeta.Hashes) != len(configTargetMeta.Hashes) {
+			return fmt.Errorf("target '%s' has %d hashes in directory repository and %d hashes in config repository", targetPath, len(targetMeta.Hashes), len(configTargetMeta.Hashes))
+		}
+		for kind, directorHash := range targetMeta.Hashes {
+			configHash, found := configTargetMeta.Hashes[kind]
+			if !found {
+				return fmt.Errorf("hash '%s' found in directory repository and not in config repository", directorHash)
+			}
+			if !bytes.Equal([]byte(directorHash), []byte(configHash)) {
+				return fmt.Errorf("directory hash '%s' is not equal to config repository '%s'", string(directorHash), string(configHash))
+			}
+		}
+		err = c.configTUFClient.Download(targetPath, &bufferDestination{})
```

Comment:
please add a comment here to explain this will do all TUF checks (timestamp/snapshot) for each repo.